addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/helpers/entities/application-role-methods.js

```diff
@@ -156,7 +156,32 @@ function newAppRoleEntity(accountEntity = {}, bucketEntity = {}, studyEntity = {
  * @param appRoleEntity The application role entity
  */
 function toRoleCfnResource(appRoleEntity, swbMainAccountId) {
-  const { name, accountId, qualifier, boundaryPolicyArn } = appRoleEntity;
+  const { name, studies, accountId, qualifier, boundaryPolicyArn, bucket, bucketKmsArn, awsPartition } = appRoleEntity;
+
+  const studyPolicy = new StudyPolicy();
+
+  _.forEach(studies, study => {
+    const { folder, kmsArn, kmsScope } = study;
+    const item = {
+      bucket,
+      awsPartition,
+      folder,
+      permission: {
+        read: isReadonly(study) || isReadwrite(study),
+        write: isReadwrite(study) || isWriteonly(study),
+      },
+    };
+
+    if (kmsScope === 'bucket') {
```

Comment:
maybe we throw an error if kmsScope is not in `['bucket', 'study']` so we know something went wrong?